Merge eScience Center and JASP versions

PabRod opened the pull request:
## Description
We've been working on enriching the template module with some common use cases (performing simple operations over manual inputs, inserting data, creating tabular outputs, creating plots, ...). The goal is to allow future module authors to adapt this template to their needs, softening the learning curve.

The current status is definitely work in progress. Said this, it is a good time to merge what we have done until now. I expect this template to keep growing (and at a faster rate) in the near future.

JorisGoosen commented:
Was removing the github workflows on purpose?

Cause we use those right? For updating translations etc?

JorisGoosen commented:
@FBartos @vandenman either of you have time to have a quick look at this too?

JorisGoosen commented:
One tiny nitpicky point I noticed is that QML files are supposed to start with a capital letter. This works, but it is probably better to have the example give the right example ;)

JorisGoosen commented on renv/activate.R:
```diff
@@ -0,0 +1,1220 @@
+
+local({
```
Should this file be here?

PabRod replied:
Good question. Do we have some guidelines about how to use `renv`?

vandenman replied:
yes, it should be there, otherwise renv cannot bootstrap itself.

JorisGoosen reviewed (changes requested):
Mostly looks good to me, except for the nitpicking on the capitals.
Could you change that?

PabRod commented:
> Was removing the github workflows on purpose?
> 
> Cause we use those right? For updating translations etc?

I kept the GitHub workflows out of this because @recap was actively working on it, and he is currently out of office. If you want to deal with this in the same PR, just let me know and I'll recover them.

PabRod commented:
> Mostly looks good to me, except for the nitpicking on the capitals. Could you change that?

Yes, and I don't think it's nitpicking. We want our template to be, as much as possible, an example of best practices :)

JorisGoosen commented:
> > Was removing the github workflows on purpose?
> > Cause we use those right? For updating translations etc?
> 
> I kept the GitHub workflows out of this because @recap was actively working on it, and he is currently out of office. If you want to deal with this in the same PR, just let me know and I'll recover them.

Ok cool, but then dont delete them outright, just use the versions that we already have.

JorisGoosen commented on inst/Description.qml:
```diff
@@ -5,10 +5,55 @@ Description
+
+	Analysis
+	{
+	  title: "Add one"      // Title for window
+	  menu: "Add one"       // Title for ribbon
```
So many hidden features :p

Sorry about my bad docs, I guess some stereotypes are true

JorisGoosen commented on NAMESPACE:
Ah yes, that is a good one.

PabRod replied:
I'm a big fan of `roxygen2`, to the point I felt dirty by editing the NAMESPACE by hand.

Said this, I'd say this deserves its own independent issue. I took the liberty of [creating it](https://github.com/jasp-stats/jaspModuleTemplate/issues/13).

PabRod commented on README.md:
```diff
@@ -1 +1,11 @@
+
+## Contributing back new module to JASP
+
+Once you have developed your module, you can contribute it back to JASP by creating a pull request. The JASP team will review your module and provide feedback. Once your module is accepted, a new repo is created in the JASP organization and your module is added to the JASP module repository.
```
That will be explained in the near future. For now, I suggest keeping this as a separate issue.

PabRod commented on tests/testthat.R:
```diff
@@ -1,4 +1,2 @@
 library(jaspTools)
 library(testthat)
-
-jaspTools::runTestsTravis(module = getwd())
```
Reverted (cf. https://github.com/jasp-stats/jaspModuleTemplate/pull/12/commits/97d3834544d4332cff4545541659b4e3bdd78d8f).

FBartos commented on R/examples.R:
```diff
@@ -0,0 +1,54 @@
+  # Dataset access
+  # options$ts --maps to--> 't'
+  # dataset[, options$ts] --maps to--> dataset$t
+  result <- paste(dataset[, options$ts], collapse = "")
```
also, I thought that `dataset[[options[["ts"]]]]` should be preferred to avoid the potential partial matching issues...
In addition, it greatly simplifies options renaming if needed later on.

PabRod replied:
Done!
Thanks!

a4603c8

FBartos commented on R/examples.R:
```diff
@@ -0,0 +1,54 @@
+          ggplot2::xlim(-3, 3) +
+          ggplot2::ylim(0, 10) +
+          ggplot2::geom_function(fun = f)
+
```
it would be good to add the usual jasp styling to to figure

```suggestion
# add jasp theme
p <- jaspGraphs::geom_rangeframe() +
    jaspGraphs::themeJaspRaw()
```

FBartos commented on R/examples.R:
```diff
@@ -0,0 +1,54 @@
+                                 width = 160,
+                                 height = 320)
+  parabolaPlot$dependOn(c("a")) # Refresh view whenever a changes
+
```
According to @JorisGoosen this should automatically add a help file entry to the figure. Could you add this to all output objects? It would be very good practice for people to document the tables/figures/html outputs that they create. The current modules are bit lacking on this, but we are trying to promote good practices here I think :)

```suggestion
parabolaPlot$info <- gettext("This figure displays a parabola specified via the `a` option.")
```

FBartos commented on R/examples.R:
```diff
@@ -0,0 +1,54 @@
+AddOne <- function(jaspResults, dataset, options) {
```
I thought the style guide says camelCase: 
this should be `addOne`

PabRod replied:
Done!
78eb154

FBartos commented on inst/qml/data.qml:
```diff
@@ -0,0 +1,40 @@
+  VariablesForm
+  {
+    AvailableVariablesList { name: "allVariables" }
+    AssignedVariablesList  { name: "ts"; label: qsTr("Times (t)"); singleVariable: true; allowedColumns: ["scale"] }
```
this and all the following inputs are missing documentation, probably splitting this into multiple lines would make it easier to add it, e.g.,
```suggestion
    AssignedVariablesList  {
      name: "ts"
      label: qsTr("Times (t)")
      info: qsTr("This info entry adds documentation to the (i) icon in the analysis file. E.g., Specify variable containing the time.")
      singleVariable: true
      allowedColumns: ["scale"]
    }
```

FBartos commented on inst/qml/data.qml:
```diff
@@ -0,0 +1,40 @@
+
+  Text
+  {
+      text: "This example shows how to load a dataset and perform a simple operation on it"
```
```suggestion
      text: qsTr("This example shows how to load a dataset and perform a simple operation on it")
```

FBartos reviewed (changes requested):
I also left a couple of comments, it would be very nice if the template already have documentation entries so people know how to automatically generate the help files/

PabRod commented on R/examples.R:
```diff
@@ -0,0 +1,54 @@
+  stats$addColumnInfo(name = "xs")
+
+  stats[["times"]] <- dataset[, options$ts]
+  stats[["xs"]] <- dataset[, options$xs]
```
Done!
Thanks!

a4603c8

PabRod commented:
> One tiny nitpicky point I noticed is that QML files are supposed to start with a capital letter. This works, but it is probably better to have the example give the right example ;)

Done!
1837b43

PabRod commented:
Dear @JorisGoosen, @vandenman and @FBartos, I addressed all your comments. Let me know if we are good enough to merge. After that, I'll keep working on a `develop` branch within this repo.

FBartos reviewed (approved):
Thanks!

JorisGoosen reviewed (approved):
looks fine to me

PabRod commented:
One last question. Anyone knows why `translations.yml` is failing (and email-bombing everybody, sorry for that)?

I (temporarily) removed it in the very last commit (https://github.com/jasp-stats/jaspModuleTemplate/pull/12/commits/bfb36a0362d127fff9d9866ddcedcf0c571259a0).

JorisGoosen commented:
@boutinb ?

PabRod commented:
@JorisGoosen, @RensDofferhoff, this is ready to merge. Stereotypically, I'm going on holiday tomorrow 😅. Feel free to squash and merge now or, if you prefer, wait until I'm back this Tuesday.
